lib/D2L.Extensibility.AuthSdk.UnitTests/FunctionalApiUriTests.cs

```diff
@@ -288,6 +288,32 @@ public class FunctionalApiUriTests {
 		}
 
 		[Test]
+		public void UserContext_CreateAuthUri_QuerystringIsMaintained() {
+			Uri queryUrl = m_userContext.CreateAuthenticatedUri( TestConstants.API_PATH_WITH_QUERY, "GET" );
+
+			Assert.IsTrue(
+				queryUrl.ToString().Contains( TestConstants.API_PATH_QUERYSTRING ),
+				"The querystring parameter was not present"
+				);
+		}
+
+		[Test]
+		public void UserContext_CreateAuthUri_MultipleQuerystringsAreNotPresent() {
+			const string querystringCharacterEncoded = "%3F";
+
+			Uri queryUrl = m_userContext.CreateAuthenticatedUri( TestConstants.API_PATH_WITH_QUERY, "GET" );
+			bool duplicatedQuery = (

```

Comment:
Okay, right, forgot we added query params.

However I still see the test as doing more of "does not encode the query that was on the path", which the test above already has "includes an unencoded query from the path".

One thing the test above doesn't test, is the potential for ?x_a becoming encoded I guess.

Perhaps something which tests a bit more explicitly could be worthwhile?
```csharp
[Test]
public void UserContext_CreateAuthUri_MergesQuerystringFromPath() {
  Uri authedUrl = m_userContext.CreateAuthenticatedUri( "/foo?bar=baz", "GET" );
  var query = HttpUtility.ParseQueryString( authedUrl.Query );

  Assert.Equal( "baz", query["bar"] );
  Assert.IsNotNull( query["x_a"] );
  Assert.IsNotNull( query["x_b"] );
  ...
}
```

Seems a bit better to me, in terms of actually specifying the outcome we're expecting. Only downside is that it respecifies the x_ params, but I think that's okay.